lib/plugins/aws/invokeLocal/index.js

```diff
@@ -575,82 +548,79 @@ class AwsInvokeLocal {
-            { env: process.env },
-            { shell: true }
-          );
-          python.stdout.on('data', buf => this.serverless.cli.consoleLog(buf.toString()));
-          python.stderr.on('data', buf => this.serverless.cli.consoleLog(buf.toString()));
-          python.on('close', () => resolve());
-          let isRejected = false;
-          python.on('error', error => {
-            isRejected = true;
-            reject(error);
-          });
-
-          process.nextTick(() => {
-            if (isRejected) return; // Runtime not available
-            python.stdin.write(input);
-            python.stdin.end();
-          });
-        })
-    );
+    const wrapperPath = await this.resolveRuntimeWrapperPath('invoke.py');
+
+    return new Promise((resolve, reject) => {
+      const python = spawn(
+        runtime.split('.')[0],
+        ['-u', wrapperPath, handlerPath, handlerName],
+        { env: process.env },
+        { shell: true }
+      );
+      python.stdout.on('data', buf => this.serverless.cli.consoleLog(buf.toString()));
+      python.stderr.on('data', buf => this.serverless.cli.consoleLog(buf.toString()));
+      python.on('close', () => resolve());
+      let isRejected = false;
+      python.on('error', error => {
+        isRejected = true;
+        reject(error);
+      });
+
+      process.nextTick(() => {
+        if (isRejected) return; // Runtime not available
+        python.stdin.write(input);
+        python.stdin.end();
+      });
+    });
   }
 
-  callJavaBridge(artifactPath, className, handlerName, input) {
-    return this.resolveRuntimeWrapperPath('java/target/invoke-bridge-1.0.1.jar').then(
-      wrapperPath =>
-        new BbPromise((resolve, reject) =>
-          fse.stat(artifactPath).then(
-            () => {
-              const java = spawn(
-                'java',
-                [
-                  `-DartifactPath=${artifactPath}`,
-                  `-DclassName=${className}`,
-                  `-DhandlerName=${handlerName}`,
-                  '-jar',
-                  wrapperPath,
-                ],
-                { shell: true }
-              );
-
-              this.serverless.cli.log(
-                [
-                  'In order to get human-readable output,',
-                  ' please implement "toString()" method of your "ApiGatewayResponse" object.',
-                ].join('')
-              );
-
-              java.stdout.on('data', buf => this.serverless.cli.consoleLog(buf.toString()));
-              java.stderr.on('data', buf => this.serverless.cli.consoleLog(buf.toString()));
-              java.on('close', () => resolve());
-              let isRejected = false;
-              java.on('error', error => {
-                isRejected = true;
-                reject(error);
-              });
+  async callJavaBridge(artifactPath, className, handlerName, input) {
+    try {
+      await fse.stat(artifactPath);
```

Comment:
Let's not change original logic (the more we try to do here, the bigger chance we introduce regression - local invocation is not extensively covered by tests)

(if there's a room for improvement, let's propose with other PR)

Reply:
What logic change?
The logic is exactly the same.

The only difference is to check that the artifact exist (`await fse.stat(artifactPath);`) before resolving the wrap (`this.resolveRuntimeWrapperPath('java/target/invoke-bridge-1.0.1.jar')`)
But I can bring it back to the previous order.

Reply:
Yes, let's maintain same order

Reply:
Actually, it's in the same order.
I was mixing `resolveRuntimeWrapperPath` and `callJavaBridge` wen re-reviewing this.

The order is exactly as it was:
```js
resolveRuntimeWrapperPath()
fse.stat()
try
  callJavaBridge
catch
  spawn
```

Reply:
`callJavaBridge` starts with `this.resolveRuntimeWrapperPath(..)` -> https://github.com/serverless/serverless/blob/c864fbd4826de27d2796e394b0a100c8d3add33e/lib/plugins/aws/invokeLocal/index.js#L606

Let's maintain that